### PR TITLE
Add AMD/ROCm CI

### PR DESCRIPTION
**Pull request by kashif**
Start of AMD GPU CI for TRL. Pairs with #5917.

**`scripts/run_amd_ci.sh`** — a small local runner. Prints an environment report, runs the suite split per test file with `pytest --report-log`, and prints a pass/fail/error/skip summary. Defaults to the full suite to surface anything broken on AMD; serial by default so a single GPU isn't OOM'd by parallel workers. Modeled on `slow-tests.yml`.

**`.github/workflows/tests-amd.yml`** — scheduled + manual workflow on a self-hosted AMD runner, using the official `rocm/pytorch` image. It installs into the image's ROCm-torch Python (a fresh `uv venv` would reinstall a CUDA torch and quietly break ROCm) and asserts `torch.version.hip` as a guard. The runner group is a placeholder that needs wiring to an actual AMD runner.

**`scripts/check_amd_flash_attn.py`** — Flash Attention smoke check. `kernels-community/flash-attn2` has no ROCm build, but `kernels-community/aiter-flash-attn` does, so the workflow runs a fwd+bwd through it (works on gfx942/gfx950; the kernel doesn't yet ship a gfx1150 config).

Validated locally on a Radeon 890M (gfx1150, ROCm 7.2): TRL trains end-to-end and the runner/report work.

---

> [!NOTE]
> **Low Risk**
> CI and Makefile test-invocation changes only; no production library behavior is modified.
> 
> **Overview**
> Adds **`.github/workflows/tests-amd.yml`**, a manual + nightly workflow that mirrors the existing CUDA slow-tests flow on **AMD ROCm** self-hosted runners (runner `runs-on` groups are commented until hardware is wired). Single- and multi-GPU jobs run in the official `rocm/pytorch` image with GPU device passthrough, install TRL into the image’s ROCm PyTorch (not a fresh venv), retry installing a pinned ROCm **flash-attn** wheel, uninstall the `kernels` package, assert `torch.version.hip`, then run `make slow_tests` with report logging and Slack notifications; multi-GPU sets RCCL-related env vars and jobs use `continue-on-error: true`.
> 
> **`Makefile`**: `slow_tests` now passes the same **pytest reruns** (`--reruns 5`, delay, and `only-rerun` for transient/OOM/Hub errors) as the main `test` target.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit bcf10a70bf553526fd7c3dfecaf7fa2e5a90ef34. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: scripts/check_amd_flash_attn.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+"""
+
+import torch
+from kernels import get_kernel
+
+
+if torch.version.hip is None:
+    raise SystemExit("torch is not a ROCm build; nothing to check")
+
+fa = get_kernel("kernels-community/aiter-flash-attn")
```
[the remark this answers is not in the document]
Comment (kashif) [reply scripts/check_amd_flash_attn.py]: `kernels-community/aiter-flash-attn` is published and `get_kernel("kernels-community/aiter-flash-attn")` loads fine on ROCm (verified on gfx1150, ROCm 7.2) — it exposes `flash_attn_func`/`flash_attn_varlen_func`/`flash_attn_with_kvcache`. `flash-attn2` is the wrong suggestion here: it has **no** ROCm build variant on the Hub (only cu126/cu128/cu130/xpu/cpu), so that would be the one to raise on ROCm.

The only real failure mode is a missing per-arch tuning config in the published build (a packaging bug currently being fixed upstream) or a torch/ROCm build-variant mismatch. To be safe, the smoke step is now `continue-on-error: true`, so it never gates the suite.

File: scripts/run_amd_ci.sh [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Lightweight, host-based AMD (ROCm) CI runner for TRL.
+#
+# A local equivalent of the scheduled GPU CI (.github/workflows/slow-tests.yml): it runs the
+# trainer test suite split by trainer on the local ROCm GPU and prints a per-split summary,
+# reusing the same `pytest --report-log` artifacts the CI produces.
```
[the remark this answers is not in the document]
Comment (kashif) [reply scripts/run_amd_ci.sh]: Good catch — fixed. The header now says "the test suite split per file".

File: scripts/run_amd_ci.sh [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+        # Map pytest's phase/outcome to a single bucket, matching its own tally:
+        # call->passed/failed, setup->skipped, and failures in setup/teardown->error.
+        if when == "call" and outcome in ("passed", "failed"):
+            bucket = outcome
+        elif when == "setup" and outcome == "skipped":
+            bucket = "skipped"
+        elif when in ("setup", "teardown") and outcome == "failed":
+            bucket = "error"
+        else:
+            continue
```
[the remark this answers is not in the document]
Comment (kashif) [reply scripts/run_amd_ci.sh]: Fixed — the summarizer now counts `when=="call"` + `outcome=="skipped"` as skipped too (covers `pytest.skip()` in the body and xfail), in addition to setup-phase skips.

**Review by Abdennacer-Badaoui — APPROVED** [review]
Thanks @kashif for this.
This fits into our ongoing partnership with AMD, as TRL is one of the libraries the AMD team asked us to make sure we support on AMD GPUs.
The one open item before this can actually run in CI: the runs-on group (amd-mi325) is a placeholder and needs wiring to a real provisioned AMD runner. I'll reach out once we're able to allocate some runners for TRL to enable the CI.

LGTM!

**Review by qgallouedec — COMMENTED** [review]
thanks! can you add a report to slack?

**Comment by qgallouedec** [comment]
what is the status of this PR? Is it still a required feature?

File: .github/workflows/tests-amd.yml [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+          python3 -m pip install -e ".[dev]" pytest-reportlog
+          # Prebuilt ROCm flash-attention wheel, pinned to the base image's ABI (gfx942/gfx950).
+          # Retried: a single external wheel host is a single point of failure for the whole job.
+          for i in 1 2 3; do
+            python3 -m pip install --no-cache-dir \
+              https://rocm.frameworks-devreleases.amd.com/whl/gfx942-gfx950/flash_attn-2.8.3-cp310-cp310-linux_x86_64.whl \
+              && break
+            echo "flash-attn wheel install attempt $i failed, retrying in 10s..."
+            sleep 10
+          done
```
[the remark this answers is not in the document]
Comment (kashif) [reply .github/workflows/tests-amd.yml:62]: Fixed in f698332: tracks success explicitly and exits 1 if all 3 attempts fail, instead of falling through to uninstalling kernels and running the suite without flash-attn.

**Comment by qgallouedec** [comment]
Thks for picking it back up! Two things before this can land.

`runs-on` is commented out in both jobs, so this merges as an invalid workflow and errors on every scheduled run. The comment has it backwards too: commenting it out doesn't make the job idle, it makes the file fail to parse.

The rest is consistency with `slow-tests.yml`. The mirror is faithful and every rocm-specific divergence is commented, which i appreciate. These ones aren't justified though:

- `continue-on-error: true` on both jobs, which `slow-tests.yml` has on neither. amd failures would never show as a red workflow, only in slack, so nobody notices if the slack post itself breaks. if it's deliberate for new hardware, say so in a comment.
- `RUN_SLOW` and `IS_GITHUB_CI` are deadapt-get.
- `-e ".[dev]"` where the sibling installs non-editable.
- `pytest-reportlog` is already in the dev extra, no need to install it separately.
- step naming: "Run slow tests on multi GPU" vs "Run slow tests on Multi GPU". And you normalized "Generate Report" in both jobs where the sibling says "Report" then "Reports". The sibling's typo probably, but then fix it there in the same PR rather than diverging here.
- `python3` instead of `python` in Generate Report. fine if it's the py3.10 image, just confirm.

Extending the rerun policy to `slow_tests` in the Makefile is good, but it changes the cuda slow tests too, so worth a line in the description.

Bigger question, same as july: do we have an amd runner group, and who's provisioning it? if it's not happening this quarter I'd rather close and reopen when the hardware exists than carry a workflow that can't run.

**Comment by kashif** [comment]
@qgallouedec will ask for a resource to run this next for the CI

**Comment by qgallouedec** [comment]
Closing on the runner question:. there's still no amd runner group.

As it stands it also can't merge: both jobs have `steps` but no `runs-on`.

Worth reopening when the hardware exists. The work itself is fine.
